multiuser/permission/che-multiuser-permission-workspace/src/main/java/org/eclipse/che/multiuser/permission/workspace/server/spi/jpa/MultiuserJpaWorkspaceDao.java

```diff
@@ -311,6 +314,21 @@ public void onCascadeEvent(BeforeAccountRemovedEvent event) throws Exception {
               (maxItems, skipCount) ->
                   workspaceManager.getByNamespace(
                       event.getAccount().getName(), false, maxItems, skipCount))) {
+        // Will try to stop running workspaces
+        if (WorkspaceStatus.RUNNING.equals(workspace.getStatus())) {
```

Comment:
What about other statuses e.g. `STARTING` ?

Reply:
Also, coud you please extract this code to the separate method with the java doc

Reply:
done